ale/base/label_isis.py

```diff
@@ -190,12 +198,20 @@ def spacecraft_clock_stop_count(self):
         : str
           Spacecraft clock stop count
         """
-        if 'SpacecraftClockStopCount' in self.label['IsisCube']['Instrument']:
-            return self.label['IsisCube']['Instrument']['SpacecraftClockStopCount']
-        elif 'SpacecraftClockStopCount' in self.label['IsisCube']['Archive']:
-            return self.label['IsisCube']['Archive']['SpacecraftClockStopCount']
-        else:
-            return None
+        if not hasattr(self, "_clock_stop_count"):
+            if 'SpacecraftClockStopCount' in self.label['IsisCube']['Instrument']:
+                self._clock_stop_count = self.label['IsisCube']['Instrument']['SpacecraftClockStopCount']
+            elif 'SpacecraftClockStopCount' in self.label['IsisCube']['Archive']:
+                self._clock_stop_count = self.label['IsisCube']['Archive']['SpacecraftClockStopCount']
+            else:
+                self._clock_stop_count = None
+
+            if isinstance(self._clock_stop_count, pvl.Quantity):
+                self._clock_stop_count = self._clock_stop_count.value
+
+            self._clock_stop_count = str(self._clock_stop_count)
```

Comment:
Does it make sense to have a generic accessor for start/stop times and then call that from these properties with an appropriate arg? Looks like a fair bit of duplicative code, but perhaps that is more readable and preferable?

Reply:
This code is part of the generic start/stop time accessors. There are a variety of different ways stop times are computed/recorded on the labels. This property handles the spacecraft clock stop count which can then be turned into the stop time by a NAIF call. See [NaifSpice::ephemeris_stop_time](https://github.com/USGS-Astrogeology/ale/blob/master/ale/base/data_naif.py#L495).

We could write a generic label search function. There is a lot of repeated code in this class to handle the multiple different places in a label a value can appear.

Reply:
Agree with Jesse here, it also may not be worth it as accessing some of these values is so specific that a generic function could only work in a few instances.

It may be worth exploring but some duplicate code is so small that maintaining a function that gets used in like two places may be just as annoying to maintain as the duplicate code